### PR TITLE
#2541 SineWave model added, descriptions corrected

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/package.order [hunk 1]
```diff
@@ -1,3 +1,4 @@
+SwitchOff
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/package.order:1]: SwitchOff is a package saved as one file. Usually packages are placed before models and blocks in a list.

File: dynawo/sources/Models/Modelica/PreassembledModels/SineWave.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, you can obtain one at http://mozilla.org/MPL/2.0/.
+    SPDX-License-Identifier: MPL-2.0
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source suite
+    of simulation tools for power systems.
+-->
+<dyn:dynamicModelsArchitecture xmlns:dyn="http://www.rte-france.com/dynawo">
+  <dyn:modelicaModel id="SineWave">
+    <dyn:unitDynamicModel id="sineWave" name="Dynawo.Electrical.Controls.Basics.SineWave"/>
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/PreassembledModels/SineWave.xml:16]: Indeed.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave_INIT.mo [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools
+* for power systems.
+*/
+
+model SineWave_INIT "Initialization for the sine wave model"
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave_INIT.mo:16]: No, it is useless, like SetPoint_INIT.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/package.order (hunk 1 above)
[the remark this answers is not in the document]
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/package.order:1]: ok

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave_INIT.mo (hunk 3 above)
[the remark this answers is not in the document]
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave_INIT.mo:16]: Then we can discard it

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+model SineWave "Parameterizable sine wave model"
+  import Dynawo.Connectors;
+
+  Connectors.ImPin source(value(start = Value0));
+
+  parameter Real K "Amplitude of the sine wave";
+  parameter Types.AngularVelocity Omega "Pulsation of the sine wave";
+  parameter Types.Angle Phi "Phase of the sine wave";
+
+  parameter Real Value0 "Start value of the sine wave model";
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo:25]: parameter Real Offset "Average value of the sine wave model";

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* for power systems.
+*/
+
+model SineWave "Parameterizable sine wave model"
+  import Dynawo.Connectors;
+
+  Connectors.ImPin source(value(start = Value0));
+
+  parameter Real K "Amplitude of the sine wave";
+  parameter Types.AngularVelocity Omega "Pulsation of the sine wave";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo:22]: import Types missing, no ?

**Review by cossartque — CHANGES_REQUESTED** [review]

**Comment by cossartque** [comment]
Also, it seems that build fails

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo (hunk 5 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SineWave.mo:22]: In the encapsulated package Dynawo, there is 

import Dynawo.Types;

So this line is not needed in the package.

**Pull request merged**